include/utils/config.hpp

```diff
@@ -85,7 +85,7 @@ class Config {
     /// \brief creates a new Config object, looking for the config.json and schemes folder relative to the provided \p
     /// main_dir
     explicit Config(std::string schemas_dir, std::string config_file, std::string modules_dir,
-                    std::string interfaces_dir, std::string types_dir);
+                    std::string interfaces_dir, std::string types_dir, std::string user_config_file);
```

Comment:
would probably be best to change it to (boost/std)::filesystem paths

Reply:
That's another point, but these should also be const refs :laughing: 
I'm really sure if these would need to be filesystem paths in case of the constructor, imagine one day you want to support something like config_file="http;//link.to/config_file"
maybe we should keep them as strings and internally complain if they can't be found

Reply:
Sure, let's change them to const refs and keep them as strings to leave that option open for the future :)